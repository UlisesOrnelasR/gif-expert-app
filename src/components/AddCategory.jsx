import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  const onSubmit = (eve) => {
    eve.preventDefault();

    if (inputValue.trim().length <= 1) {
      return;
    }
    //setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue("");
  };
  return (
    <form onSubmit={(eve) => onSubmit(eve)}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={(eve) => onInputChange(eve)}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
