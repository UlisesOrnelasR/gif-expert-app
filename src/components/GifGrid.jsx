import React from "react";

export const GifGrid = ({ category }) => {
  return <div>GifGrid</div>;
};
