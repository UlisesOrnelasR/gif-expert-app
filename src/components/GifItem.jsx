export const GifItem = ({ image }) => {
  console.log(image);
  return (
    <>
      <h1>fff</h1>
    </>
  );
};
