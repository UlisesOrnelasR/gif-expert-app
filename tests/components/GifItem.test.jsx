import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebass sobre el componente <GifItem />", () => {
  const title = "Naruto";
  const url = "https://naruto.com";
  test("Deberia de hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test("Deberia mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    // expect(screen.getByRole("img").alt).toBe(title);
    const { alt } = screen.getByRole("img");
    expect(alt).toBe(alt);
  });
  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy;
  });
});
