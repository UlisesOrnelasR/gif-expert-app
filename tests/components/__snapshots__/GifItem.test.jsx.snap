// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebass sobre el componente <GifItem /> Deberia de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto.com"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
